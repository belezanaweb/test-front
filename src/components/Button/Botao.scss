@import '../../assets/sass/colors';

.botao {
  width: 100%;
  font-weight: 700;
  font-size: 20px;
  letter-spacing: 0.5px;
  height: 60px;
  font-size: 20px;
  font-weight: 700;
  color: #fff;
  background: $primaryColor;
  display: block;
  text-transform: uppercase;
  border: 0;
  box-shadow: 0 2px 4px rgba(0, 0, 0, .25), #D45A00 0 -3px 0 inset;
  border-radius: 3px;
}
