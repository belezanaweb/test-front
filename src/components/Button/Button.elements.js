import styled from 'styled-components'

export const ButtonStyled = styled.button`
  cursor: pointer;
  border: none;
  outline: none;
  font-size: 1.25rem;
  line-height: 1.5rem;
  letter-spacing: 0.5px;
  text-align: center;
  width: 100%;
  height: 3.75rem;
  color: #fff;
  border-radius: 3px;
  background-color: #ff6c00;
  box-shadow: 0 3px 0 #d45a00, 0 2px 4px rgba(0, 0, 0, 0.25);

  &:hover {
    background-color: #d45a00;
  }
`
