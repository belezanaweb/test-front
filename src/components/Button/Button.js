import React from 'react'

const Button = () => {
  return (
    <div>
      <button></button>
    </div>
  )
}

export default Button
