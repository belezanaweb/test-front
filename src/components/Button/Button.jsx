import React from 'react'

import { StyledButton } from './styles'

function Button({ children, ...rest }) {
  return <StyledButton {...rest}>{children}</StyledButton>
}

export default Button
