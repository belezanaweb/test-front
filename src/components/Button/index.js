import React from 'react'
import { StyledButton } from './styles'

const Button = ({ title, onClick, ...props }) => {
  return (
    <StyledButton {...props} onClick={onClick}>
      {title}
    </StyledButton>
  )
}

export default Button
