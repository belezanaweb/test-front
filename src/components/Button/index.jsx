import React from 'react'
import * as S from './styled'
import { useNavigate } from 'react-router-dom'

const Button = ({ text, path, disabled }) => {
  let navigate = useNavigate()
  return (
    <S.Button onClick={() => navigate(`${path}`)} disabled={disabled}>
      {text}
    </S.Button>
  )
}

export default Button
