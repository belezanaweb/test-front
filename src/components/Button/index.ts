import styled from 'styled-components'

export const Button = styled.button`
  width: 100%;
  background-color: ${(props) => props.theme.light.colors['--c-contrast-two']};
  border: none;
  padding: 15px 0;
  border-radius: 3px;

  box-shadow: inset 0 -3px 0 0 ${(props) => props.theme.light.colors['--c-contrast-tree']},
    1px 1px 10px 1px rgb(0 0 29 / 12%);

  color: white;
  font-size: ${(props) => props.theme.fontSizes.extraLarge};
  font-weight: bold;
  text-transform: uppercase;
  text-decoration: none;
  text-align: center;

  &:hover {
    background-color: ${(props) => props.theme.light.colors['--c-contrast-tree']};
  }

  &:disabled {
    background-color: ${(props) => props.theme.light.colors['--c-contrast-tree']};
  }
`
