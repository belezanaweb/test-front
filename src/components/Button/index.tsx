import React from 'react';
import { Container } from './styles';

const Button: React.FC<React.ButtonHTMLAttributes<HTMLButtonElement>> = ({
  children,
  className,
  role = 'button',
  ...rest
}) => {
  return (
    <Container className={className} role={role} {...rest}>
      {children}
    </Container>
  );
};

export default Button;
