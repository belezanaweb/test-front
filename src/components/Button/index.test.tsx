import { render, fireEvent, screen } from '@testing-library/react';
import React from 'react';
import Button from '.';

describe('<Button />', () => {
  it('should be able to render correctly', () => {
    // Arrange
    const { baseElement } = render(<Button>Button</Button>);

    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  it('should be able to render correctly when mouse enter', () => {
    // Arrange
    const { baseElement } = render(<Button>Button</Button>);

    // Act
    fireEvent.mouseEnter(screen.getByText('Button'));

    // Assert
    expect(baseElement).toMatchSnapshot();
  });

  it('should be able to render correctly when disabled', () => {
    // Arrange
    const { baseElement } = render(<Button disabled>Button</Button>);

    // Assert
    expect(baseElement).toMatchSnapshot();
  });
});
