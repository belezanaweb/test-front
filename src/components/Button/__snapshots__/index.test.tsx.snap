// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Button /> should be able to render correctly 1`] = `
<body>
  <div>
    <button
      class="sc-bdnylx gAlZQz"
      role="button"
    >
      Button
    </button>
  </div>
</body>
`;

exports[`<Button /> should be able to render correctly when disabled 1`] = `
<body>
  <div>
    <button
      class="sc-bdnylx gAlZQz"
      disabled=""
      role="button"
    >
      Button
    </button>
  </div>
</body>
`;

exports[`<Button /> should be able to render correctly when mouse enter 1`] = `
<body>
  <div>
    <button
      class="sc-bdnylx gAlZQz"
      role="button"
    >
      Button
    </button>
  </div>
</body>
`;
